rust/src/dns/dns.rs

```diff
@@ -115,6 +115,36 @@ pub const DNS_RCODE_BADTRUNC: u16 = 22;
 
 static mut ALPROTO_DNS: AppProto = ALPROTO_UNKNOWN;
 
+/// Trait for objects to implement methods that calculate the size of that object.
+pub trait SizeOf {
```

Comment:
if we define a trait like this, do we risk name collisions with future rust features?

Reply:
These name collisions happen more with keywords.  Traits are name spaced like function names and very few are brought into the global name space (https://doc.rust-lang.org/std/prelude/index.html), but even then its not a compile error to redeclare a trait, the scoping rules will determine what has precedence, so its not something I'd worry about until it became an issue.